proto/momento/auth.proto

```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "momento.auth";
+
+package momento.auth;
+
+service Auth {
+  rpc Login (_LoginRequest) returns (stream _LoginResponse) {}
+}
+
+message _LoginRequest {
+}
+
+message _LoginResponse {
+  oneof state {
+    DirectBrowser direct_browser = 1;
+    LoggedIn logged_in = 2;
+    Message message = 3;
+    Error error = 4;
```

Comment:
How do I get into the DirectBrowser or Message state during a login? Are those intermediary states on users path  to LoggedIn status? Just confused how I get into either of those states here. LoggedIn or Error seems clear to me.

Reply:
These are tools for SDK writers to implement the login flow. I will consume these states in `momento`. Their semantics are briefly described on the respective messages' definitions.

Each of these messages is a command from the service telling the login process on the client machine what to do.
* DirectBrowser tells the client machine to... direct a browser to the url.
* LoggedIn tells the client machine that the user has logged in (and gives it the session token).
* Message delivers a message destined for the user (implemented, e.g., via printf)
* Error tells the client that something went wrong, and a description of what.
* The end of the grpc `stream` response indicates that the login is done. LoggedIn and Error happen to be the 2 intended final state messages.

I don't think you're asking for an exhaustive listing of all defined oauth2 login dispositions but I'm not sure what you're looking for otherwise. I can point you to the server-side workflow implementation that is already checked in if you like?

Reply:
I see this helps! I think was just just confused on how this was used as stream back and forth. You descriptions helped me visualize more and makes sense 👍 .

Reply:
cool - yeah, login is actually is a one-way stream. The IDP flow is out-of-band and happens (at least in this iteration) via the DirectBrowser state though we might have other flows for special cases in the future.